tests/components/squeezebox/test_init.py

```diff
@@ -0,0 +1,23 @@
+"""Test squeezebox update platform."""
+
+from unittest.mock import patch
+
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+async def test_init_api_fail(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test init fail due to API fail."""
+
+    # Setup component to fail...
+    with (
+        patch(
+            "homeassistant.components.squeezebox.Server.async_query",
+            return_value=False,
+        ),
+    ):
```

Comment:
In the future, maybe its nice to check if you can do a full patch of the `Server` object. This allows for easy testing. Checkout `airgradient` for an example